src/ninja.cc

```diff
@@ -209,6 +209,10 @@ void Usage(const BuildConfig& config) {
 "  -j N     run N jobs in parallel [default=%d, derived from CPUs available]\n"
 "  -k N     keep going until N jobs fail [default=1]\n"
 "  -l N     do not start new jobs if the load average is greater than N\n"
+"  -m N     do not start new jobs if the memory usage exceeds N percent\n"
+#if !(defined(__APPLE__) || defined(linux) || defined(_WIN32))
+"           (not yet implemented on this platform)\n"
+#endif
 "  -n       dry run (don't run commands but act like they succeeded)\n"
 "  -v       show all command lines while building\n"
```

Comment:
This doesn't document the new -D and -M parameters. With those included, this increases the number of switches in ninja by 25%. Says HACKING.md: "I am very wary of changes that increase the complexity of Ninja (in
particular, new build file syntax or command-line flags)". Maybe there's a way to handle low-memory situations with less ninja flag surface?

Reply:
@nico, I agree that this needs to be re-worked.  There needs to be a single top-level parameter for constraints, similar to debug output option.
I'm currently using https://github.com/maxim-kuvyrkov/ninja/commit/4fb7ab82334efcc5cf4fc45664a632a8d9ff0813 quite successfully, but this code needs to be refactored into a loop over different constraints.  If the constraints in above commit look useful to upstream ninja, I'll clean this up together with option handling, and re-submit.